### ISSUE TITLE
FP same_functions_in_if_condition with cfg!() with macro as condition

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#6168]
I tried this code:

```rust
#![warn(clippy::same_functions_in_if_condition)]
pub fn main() {
    let os = if cfg!(target_os = "macos") {
        "macos"
    } else if cfg!(target_os = "windows") {
        "windows"
    } else {
        "linux"
    };

    println!("{}", os);
}
```

This will boil down to something like `... if false { "macos" } else if false { "windows" } else { "linux" }` on linux, the lint should probably just ignore exprs that originate from macros?

current warning:
````
warning: this `if` has the same function call as a previous `if`
 --> src/main.rs:5:15
  |
5 |     } else if cfg!(target_os = "windows") {
  |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![warn(clippy::same_functions_in_if_condition)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: same as this
 --> src/main.rs:3:17
  |
3 |     let os = if cfg!(target_os = "macos") {
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#same_functions_in_if_condition
  = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
````
### Meta

`clippy 0.0.212 (8dae8cd 2020-10-12)`
